newm-server/src/main/kotlin/io/newm/server/aws/AwsSecretsManagerConfigLoader.kt

```diff
@@ -0,0 +1,66 @@
+package io.newm.server.aws
+
+import com.amazonaws.regions.Regions
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerAsyncClientBuilder
+import com.amazonaws.services.secretsmanager.model.GetSecretValueRequest
+import com.typesafe.config.ConfigFactory
+import io.ktor.server.config.ApplicationConfig
+import io.ktor.server.config.ConfigLoader
+import io.ktor.server.config.HoconApplicationConfig
+import kotlinx.serialization.json.Json
+import java.io.File
+
+class AwsSecretsManagerConfigLoader : ConfigLoader {
+    private val secretsManager = AWSSecretsManagerAsyncClientBuilder.standard()
+        .withRegion(Regions.fromName("us-west-2"))
+        .build()
+    private val json = Json {
+        ignoreUnknownKeys = true
+        explicitNulls = false
+    }
+
+    override fun load(path: String?): ApplicationConfig? {
+        if ((path == null) || !path.startsWith("arn:aws:secretsmanager")) {
+            return null
+        }
+        val result = secretsManager.getSecretValue(
+            GetSecretValueRequest().withSecretId(path)
+        )
+        val secretsMap: Map<String, String> = json.decodeFromString(result.secretString)
+        val configString = convertToNestedHocon(secretsMap)
+
+        val file = File.createTempFile("temp", ".conf")
```

Comment:
again - more awkwardness since ktor classes expect a file. Ideally this should be done in memory or we should implement our own Config class

Reply:
Can you try to use `MapApplicationConfig` instead of `HoconApplicationConfig`? It might be less awkward.